web_chatter_paste/tests/test_web_chatter_paste.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from mock import MagicMock
+from odoo import http
+from odoo.tests.common import HttpCase
+
+
+class TestWebChatterPaste(HttpCase):
+    def test_controller(self):
+        partner_id = self.ref('base.main_partner')
+        attachment_obj = self.env['ir.attachment']
+
+        attachment_count = attachment_obj.search_count([
+            ('res_model', '=', 'res.partner'),
+            ('res_id', '=', partner_id)
+        ])
+
+        f = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAAXNSR0IArs4c6' \
+            'QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAAYdEVYdFN' \
+            'vZnR3YXJlAHBhaW50Lm5ldCA0LjAuOWwzfk4AAAAMSURBVBhXY/j//z8ABf4C/q' \
+            'c1gYQAAAAASUVORK5CYII='
+
+        request = http.request
+        http.request = MagicMock(env=self.env)
```

Comment:
This is not correctly unpatching. You should mock using with statements & patching like so:

```
with mock.patch.object(http, 'request') as request:
    request.env = self.env
    ....
```

Not doing this will leave the request object as a mock in the http module. This will then affect subsequent tests if they require a request.